File: tests/__tests__/functions/array_of_functions/array_of_functions.js

import { expect, test } from '@jest/globals';
import { runFile } from '../../../helpers/testsHelper';

let pjs = runFile(import.meta.url, 'array_of_functions.pas');
// insp(pjs.engine.tree);

test("write and writeln don't fail", () => {
  expect(pjs.getVarValue('a')).toBe(25)
});

File: tests/__tests__/functions/array_of_functions/array_of_functions.pas
var a: integer;
ar: array [1..12] of function(a, b: integer): integer;

function dataFunc(a: integer; b: integer): integer;
begin
    dataFunc := a + b;
end;

begin
   ar[8] := dataFunc;
   a := ar[8](12, 13);
   writeln(a);
end.